torchchronos/lightning/tfc_pretrain.py

```diff
@@ -0,0 +1,71 @@
+from pathlib import Path
+from lightning import LightningDataModule
+
+from torch.utils.data import DataLoader
+
+from ..utils import swap_batch_seq_collate_fn
+
+from ..transforms import Transform
+from ..typing import DatasetSplit
+from ..datasets import TFCPretrainDataset
+
+
+class TFCPretrainDataModule(LightningDataModule):
+    """See :class:`~torchchronos.datasets.TFCPretrainDataset` for more details."""
+
+    def __init__(
+        self,
+        name: str,
+        batch_size: int = 32,
+        transform: Transform | None = None,
+    ):
+        super().__init__()
+
+        self.name = name
+        self.batch_size = batch_size
+        self.transform = transform
+
+        self.cache_dir = Path(".cache") / "data"
+
+    def prepare_data(self) -> None:
+        # The test split does not matter
+        TFCPretrainDataset(self.name, self.cache_dir, "train", transform=None)
```

Comment:
I don't know what you mean by the workaround but this was deliberate. We want a dataset so we create an instance of it since that instance knows how to initialize itself. It also breifly loads the dataset but thats very fast anyways.

Reply:
Because it's not explicit and as far as I can see the only thing you do in the constructor is calling download (and checking for the cache)